03week/towersOfHanoi.js

```diff
@@ -19,23 +34,62 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
-}
-
-function isLegal() {
-  // Your code here
-
-}
+// variable to store the 3 valid entries
+const validEntry = ['a', 'b', 'c'];
 
-function checkForWin() {
-  // Your code here
+// function that verifies if the entries are true or false
+const isEntryValid = (startStack,endStack) => {
+  return validEntry.includes(startStack) && validEntry.includes(endStack);
+};
 
-}
 
 function towersOfHanoi(startStack, endStack) {
-  // Your code here
+  // two variables to store the stacks called by the user
+  const currentLocation = stacks[startStack];
+  const endLocation = stacks[endStack];
+
+  // function that determines move is valid and calls movePiece if true. 
+  function isLegal() {
+  // two variables. lastArrayItem stores the value called by the user in endStack
+  // removedItem stores the value called by the user in startStack
+    const lastArrayItem = endLocation[endLocation.length - 1];
+    const removedItem = currentLocation.pop();
+    
+    // function that pushes the removedItem onto the variable endLocation
+    function movePiece() {
```

Comment:
this should be outside of the isLegal function and called in the parent functions only if the isLegal function returns true.